llvm/lib/SYCLLowerIR/ESIMD/ESIMDVerifier.cpp

```diff
@@ -60,6 +60,8 @@ static const char *LegalSYCLFunctions[] = {
     "^sycl::_V1::exp",
     "^sycl::_V1::bit_cast<.+>",
     "^sycl::_V1::operator.+<.+>",
+    "^sycl::_V1::ext::oneapi::experimental::properties",
```

Comment:
Very good question. I did not realize that the need to add these 2 lines was signal about a potential problem.

I analyzed it a bit and as I can see it is the convenience payment for syntax sugar coming from 
a) convenient creation of 'properties' class using a number of properties following in random order
b) passing `properties` object to `block_load()` and having the default value for that argument:
```c++
esimd::properties props{ext::oneapi::experimental::alignment<16>};
auto v = block_load<float, N>(ptr, props);
```

The very first IR (after C++ FE, at ESIMDVerifier.cpp) has the real 'properties' object creation, and that object is then passed to `block_load()`
```
  %"class.sycl::_V1::ext::oneapi::experimental::properties" = type { %"class.std::tuple" }
  %"class.std::tuple" = type { i8 }
  ...

  %props2 = alloca %"class.sycl::_V1::ext::oneapi::experimental::properties", align 1
  %props2.ascast = addrspacecast ptr %props2 to ptr addrspace(4)
  ...
  call spir_func void @_ZN4sycl3_V13ext6oneapi12experimental10propertiesISt5tupleIJNS3_14property_valueINS3_13alignment_keyEJSt17integral_constantIiLi16EEEEEEEEC2IJSA_EEEDpT_(ptr addrspace(4) noundef align 1 dereferenceable_or_null(1) %props2.ascast, ptr noundef byval(%"struct.sycl::_V1::ext::oneapi::experimental::property_value") align 1 %agg.tmp.ascast.ascast) #12
    // This function calls "^sycl::_V1::ext::oneapi::experimental::detail::ExtractProperties" from ESIMDVerifier.cpp filter.
  ...
  call void @llvm.memcpy.p4.p4.i64(ptr addrspace(4) align 1 %agg.tmp2.ascast, ptr addrspace(4) align 1 %props2.ascast, i64 1, i1 false), !tbaa.struct !72
  %agg.tmp2.ascast.ascast = addrspacecast ptr addrspace(4) %agg.tmp2.ascast to ptr
  ...
  call spir_func void @_ZN4sycl3_V13ext5intel5esimd10block_loadIfLi32ENS1_6oneapi12experimental10propertiesISt5tupleIJNS6_14property_valueINS6_13alignment_keyEJSt17integral_constantIiLi16EEEEEEEEEEENSt9enable_ifIXsr3ext6oneapi12experimentalE18is_property_list_vIT1_EENS3_4simdIT_XT0_EEEE4typeEPKSJ_SH_(ptr addrspace(4) sret(%"class.sycl::_V1::ext::intel::esimd::simd") align 128 %va.ascast, ptr addrspace(4) noundef %add.ptr, ptr noundef byval(%"class.sycl::_V1::ext::oneapi::experimental::properties") align 1 %agg.tmp2.ascast.ascast) #12
```
Because that `properties` object is not used inside `block_load()`, the code related to 'properties' is removed later.